pages/chat/[[...roomId]].js

```diff
@@ -169,30 +186,112 @@ export default function Chat() {
+                    </ServiceTable>
                 </details>
                 <br />
                 <details open>
-                    <summary><h3 style={{ display: 'inline-block', marginBottom: '1rem' }}>{ t('Rooms') }</h3></summary>
-                    { otherRooms && otherRooms.map((room) => <SidebarListEntry key={room.roomId} room={room} />) }
+                    <Summary><h3>{ t('Rooms') }</h3></Summary>
+                    <ServiceTable>
+                        <ServiceTable.Body>
+                            { otherRooms && otherRooms.map((room) => {
+                                return <ServiceTable key={room.roomId}><SidebarListEntry key={room.roomId} room={room} selected={room.roomId === roomId} /></ServiceTable>;
+                            }) }
+                        </ServiceTable.Body>
+                    </ServiceTable>
                 </details>
                 <br />
-            </DefaultLayout.Sidebar>
-            { roomId && (
-                <DefaultLayout.IframeWrapper>
-                    <iframe
-                        ref={iframe}
-                        title="/chat"
-                        src={`${getConfig().publicRuntimeConfig.chat.pathToElement}/#/room/${roomId}`}
-                    />
-                </DefaultLayout.IframeWrapper>
-            ) }
+            </Sidebar>
+            <DefaultLayout.IframeWrapper>
+
+                { isLoadingIframe ? <LoadingSpinner />
+                    : iframe.current && <DefaultLayout.IframeHeader>
+                        <h2>{ matrix.rooms.get(roomId)?.name }</h2>
+                        <DefaultLayout.IframeHeaderButtonWrapper>
+                            { roomId && <>
+                                <CopyToClipboard text={roomId} />
+                                <TextButton title={t('Leave room and remove from my library')} onClick={() => handleLeave(roomId)}>
+                                    { isLeavingRoom ? <LoadingSpinnerInline /> : <Icon><DeleteBinIcon /></Icon> }
+                                </TextButton>
+                            </> }
+                            <MobileBackButton onClick={toggleRoomListView}>
+                                <Icon>
+                                    <CloseIcon />
+                                </Icon>
+                            </MobileBackButton>
+                        </DefaultLayout.IframeHeaderButtonWrapper>
+                    </DefaultLayout.IframeHeader> }
+
+                { /* The iframe can take a long time to load which is why we show a loading spinner.
+                The style hack is needed because setting 'display' to 'none' creates an error.
+                Element performs a few compatability checks before onLoaded fires.
+                Therefore, Element throws an error, saying the current browser is not supported.
+                Setting the height to 0px centers the loading spinner before the iframe is ready.
```

Comment:
hm not sure why the loading spinner didn't show up for you. 
Generally I think we really need to show a loading spinner before the iframe is ready. It just feels like a bug sometimes, otherwise.
I've opened an issue for this in the meantime #119